pkg/e2e/framework/clusterapi.go

```diff
@@ -89,103 +95,84 @@ func (f *Framework) DeployClusterAPIStack(clusterAPINamespace, actuatorImage, ac
-	// No longer affecting future deployments since it lives in a different namespace.
-
-	clusterAPIDeployment := manifests.ClusterAPIDeployment(clusterAPINamespace)
-	f.By("Scaling down apiserver deployment")
-	err = f.ScaleDeploymentDownToZero(clusterAPIDeployment)
-	f.ErrNotExpected(err)
-
-	f.By("Deleting apiserver deployment")
-	WaitUntilDeleted(func() error {
-		return f.KubeClient.AppsV1beta2().Deployments(clusterAPIDeployment.Namespace).Delete(clusterAPIDeployment.Name, &metav1.DeleteOptions{PropagationPolicy: &orphanDeletepolicy, GracePeriodSeconds: &zero})
-	}, func() error {
-		_, err := f.KubeClient.AppsV1beta2().Deployments(clusterAPIDeployment.Namespace).Get(clusterAPIDeployment.Name, metav1.GetOptions{})
-		return err
-	})
-	// Ignore the error, the deployment has 0 replicas.
-	// No longer affecting future deployments since it lives in a different namespace.
-
-	f.By("Deleting cluster api service")
-	apiService := manifests.ClusterAPIService(clusterAPINamespace)
+	f.By("Deleting cluster role")
+	clusterRoleBinding := manifests.ClusterRoleBinding(clusterAPINamespace)
 	err = WaitUntilDeleted(func() error {
-		return f.KubeClient.CoreV1().Services(apiService.Namespace).Delete(apiService.Name, &metav1.DeleteOptions{})
+		return f.KubeClient.RbacV1().ClusterRoleBindings().Delete(clusterRoleBinding.Name, &metav1.DeleteOptions{})
 	}, func() error {
-		_, err := f.KubeClient.CoreV1().Services(apiService.Namespace).Get(apiService.Name, metav1.GetOptions{})
+		_, err := f.KubeClient.RbacV1().ClusterRoleBindings().Get(clusterRoleBinding.Name, metav1.GetOptions{})
 		return err
 	})
 	f.ErrNotExpected(err)
 
-	// Even though the certs are different, only the secret name(space) and apiservice name(space) are actually used
-	certsSecret, apiAPIService, err := manifests.ClusterAPIServerAPIServiceObjects(clusterAPINamespace)
-	f.ErrNotExpected(err)
-
-	f.By("Deleting cluster api api service")
+	clusterRoleManifest := manifests.ClusterRoleManifest()
 	err = WaitUntilDeleted(func() error {
-		return f.APIRegistrationClient.Apiregistration().APIServices().Delete(apiAPIService.Name, &metav1.DeleteOptions{})
+		return f.KubeClient.RbacV1().ClusterRoles().Delete(clusterRoleManifest.Name, &metav1.DeleteOptions{})
 	}, func() error {
-		_, err := f.APIRegistrationClient.Apiregistration().APIServices().Get(apiAPIService.Name, metav1.GetOptions{})
+		_, err := f.KubeClient.RbacV1().ClusterRoles().Get(clusterRoleManifest.Name, metav1.GetOptions{})
 		return err
 	})
 	f.ErrNotExpected(err)
 
-	f.By("Deleting api server certs")
-	err = WaitUntilDeleted(func() error {
-		return f.KubeClient.CoreV1().Secrets(certsSecret.Namespace).Delete(certsSecret.Name, &metav1.DeleteOptions{})
-	}, func() error {
-		_, err := f.KubeClient.CoreV1().Secrets(certsSecret.Namespace).Get(certsSecret.Name, metav1.GetOptions{})
-		return err
-	})
-	f.ErrNotExpected(err)
+	// Do not delete the CRDs as they are always in the default namespace.
```

Comment:
can we just remove this before merging?

Reply:
It still could be a valid use case. Later, I would like to if condition this part with:
```go
if deleteCRDs {
...
}
```

and `DestroyClusterAPIStack(clusterAPINamespace, actuatorImage, actuatorPrivateKey string, deleteCRDs bool)`. So rather keeping the code in comment for now.